articles/quickstart/native/ionic3/01-login.md

```diff
@@ -0,0 +1,262 @@
+
+The `AuthService` is now accessible in the view and its `login` method can be called.
+
+```html
+<!-- src/pages/home/home.html -->
+
+<ion-header>
+  <ion-navbar>
+    <ion-title>Home</ion-title>
+  </ion-navbar>
+</ion-header>
+
+<ion-content padding>
+  <p *ngIf="auth.loading" text-center>Loading...</p>
+  <ng-template [ngIf]="!auth.loading">
+    <!-- Not loading, not logged in: show login button -->
+    <button
+      ion-button
+      block
+      color="primary"
+      *ngIf="!auth.loggedIn"
+      (click)="auth.login()">Log In</button>
+    <!-- Not loading, logged in: show profile and logout button -->
+    <ng-template [ngIf]="auth.loggedIn">
+      <ion-card *ngIf="auth.user">
+        <img [src]="auth.user.picture">
+        <ion-card-content>
+          <ion-card-title>{{ auth.user.name }}</ion-card-title>
+          <pre>{{ auth.user | json }}</pre>
+        </ion-card-content>
+      </ion-card>
+      <button
+        ion-button
+        block
+        color="danger"
+        (click)="auth.logout()">Log Out</button>
+    </ng-template>
+  </ng-template>
+</ion-content>
+```
+
+## Add Platform and Run the App
+
+You will now need to allow Ionic / Cordova to install the necessary plugins and update your `config.xml` appropriately for the platform you wish to run on.
+
+Use the following commands to add your desired platform(s) (e.g., `ios` or `android`) and run the app:
+
+```bash
+# Add platform (e.g., ios or android)
+$ ionic cordova platform add {platform}
+# Run on desired platform (e.g., ios or android)
+$ ionic cordova run {platform} --livereload
+```
+
+This will then launch your app in the local emulation environment for the platform you chose (Xcode Simulator for iOS or the Android emulator of your choice).
+
+Don't be alarmed if you _cannot authenticate_ successfully yet at this stage! There was some configuration generated by running the app that you still need to add to your Auth0 settings.
+
+## Update Auth0 Dashboard Configuration
+
+1. After executing the `run` command, your `config.xml` file should contain `<allow-navigation>` tag(s).
+2. Make note of IP address URLs from any `<allow-navigation>` tags.
+3. In your [Auth0 Dashboard](https://manage.auth0.com), go to your Ionic app client's **Settings**.
+4. Add the full allowed navigation addresses (including `http://` and ports) to the **Allowed Origins (CORS)** settings for your Auth0 client and click the **Save Changes** button.
+
+You should now be able to log into your app!
+
+## Troubleshooting
+
+### Cannot read property 'isAvailable' of undefined
+
+This means that you're attempting to test this in a browser. At this time, you'll need to run this either in an emulator or on a device.
+
+### Completely blank page when launching the app
+
+This could either mean that you've built the seed project using a different version of Ionic, or that the device where you are testing it isn't entirely supported by Ionic 3 yet. Be sure to check the console for errors.
```

Comment:
This could also be faced on iOS if you are running the app in a simulator and something else is listening on port `:8080` on the same machine. A good way to check for this is to run the application from XCode instead and check for the logs.